src/CacheInvalidator.php

```diff
@@ -47,28 +48,38 @@ class CacheInvalidator
     const INVALIDATE = 'invalidate';
 
     /**
+     * Value to check support of tag invalidation operation.
+     */
+    const TAG = 'tag';
+
+    /**
      * @var ProxyClientInterface
      */
     private $cache;
 
     /**
-     * @var EventDispatcherInterface
+     * @var TagHandler
      */
-    private $eventDispatcher;
+    private $tagHandler;
 
     /**
-     * @var string
+     * @var EventDispatcherInterface
      */
-    private $tagsHeader = 'X-Cache-Tags';
+    private $eventDispatcher;
 
     /**
      * Constructor
      *
-     * @param ProxyClientInterface $cache HTTP cache
+     * @param ProxyClientInterface $cache      HTTP cache
+     * @param TagHandler           $tagHandler A tag handler instance (optional)
      */
-    public function __construct(ProxyClientInterface $cache)
+    public function __construct(ProxyClientInterface $cache, TagHandler $tagHandler = null)
```

Comment:
@stof There’s still some discussion around whether the CacheInvalidator should function as a complete facade including tagging or not. If so, it obviously requires access to the new TagHandler.